app/(auth)/change-password/action/index.ts

```diff
@@ -0,0 +1,46 @@
+'use server'
+import { cookies } from 'next/headers'
+import { redirect } from 'next/navigation'
+import type { ChangePasswordSchema } from '@/(auth)/change-password/schema'
+import { createClient } from '@/(auth)/supabase/server'
+
+const assertVerifyOldPassword = async (
+  supabase: ReturnType<typeof createClient>,
+  data: ChangePasswordSchema
+): Promise<void> => {
+  const {
+    data: { session },
+    error
+  } = await supabase.auth.getSession()
+  if (error || !session?.user.email) {
+    throw new Error('user session is not detected')
+  }
+
+  const {
+    error: signInError,
+    data: { session: signInSession }
+  } = await supabase.auth.signInWithPassword({
+    email: session.user.email,
+    password: data.oldPassword
```

Comment:
Perhaps you don't use the old password that user input, you just use session data? I could change password with wrong old password..!

Reply:
It actually worked, but forgot to add `await` keyword on top of the call site of verifyFunc..
Besides, I replaced where I put this code.